### ISSUE TITLE
Swift Support

### ISSUE THREAD

**Issue by MatthewYork** [MatthewYork/DateTools#27]
I am about to do a full port of DateTools to Swift. Before getting started, I want to open this thread for suggestions on ways to utilize the Swift language in new and unique ways to make DateTools even easier to use.

**Comment by kbpontius** [comment]
Hey, I'm looking forward to seeing this in Swift!
Have you had any progress with this yet?

**Comment by netoleal** [comment]
Hi Matthew! Any word on using the current ObjC version into a Swift framework? Since on a framework is not allowed to use bridging header, I'm got myself stucked =/ Thanks!

**Comment by Mazyod** [comment]
It works for me with swift using bridging:

```objc
#import <DateTools/DateTools.h>
```

```swift
import DateTools
```

**Comment by asad-appifytech** [comment]
Any update on this?
When we can expect to have complete Swift version of this awesome library?

**Comment by MatthewYork** [comment]
Yes. So, now that Swift three is out, I sense they are nearing an end to breaking changes on the core. Many of the function names in beta 4 were different even from beta 3, so when Swift 3 becomes stable, I will begin approaching a new version with my colleagues.

If we had made DateTools for Swift, we would have had to remake it for Swift 2... and Swift 3. So, we are biding our time until the core frameworks settle to present a Swift solution.

**Comment by MatthewYork** [comment]
Swift rewrite has begun! Feel free to follow the swift branch and critique syntax/style/implementation/etc. We want to stay as close to Swift 3 spec as possible.

Thanks!

**Comment by jeffreyjharris** [comment]
Not certain if this helps the rewrite, but there was a port of datetools to swift here: https://github.com/codewise/ios-date-tools

**Comment by heyalexchoi** [comment]
First off, this project rules. Thanks MatthewYork and contributors for all your work. It's very much appreciated!

Second, you've probably already got this, but just in case you haven't, this document has a lot of useful guidelines for writing idiomatic Swift 3 APIs: https://swift.org/documentation/api-design-guidelines/

**Comment by MatthewYork** [comment]
Thank you for the share and the kind words! We are mostly code complete with the Swift 3 version of DateTools. It is currently housed on our Swift branch, but we will be merging to master soon. Everyone involved is currently swamped with other (professional) work, but we are close! Mainly we need to shore up packaging, such as CocoaPods, SPM, and maybe even Carthage.

Stay tuned!

**Comment by Huang-Libo** [comment]
@MatthewYork 
The Obj-C version will be abandoned?
